ES2017 preset + polyfills w/ tests

bettiolo opened the pull request:
Added support for ES2017 syntax and polyfills

zloirock commented on packages/babel-preset-es2017/test/es2017.test.js:
```diff
@@ -0,0 +1,61 @@
+      [Symbol('foo')]: 123,
+    };
+    assert.deepEqual(Object.getOwnPropertyDescriptors(obj), {
+      [Symbol('foo')]: {
```
`Symbol('foo') !== Symbol('foo')`

zloirock commented:
As I wrote - polyfills in presets can cause too many problems.

bettiolo commented:
@zloirock agreed, I made it optional: 

https://github.com/babel/babel/pull/3642/files#diff-5d121dc3c95aadd75908213e61bd1bb3R1

bettiolo commented:
Changed to babel-style fixtures but I am getting an error about the unknown syntax. I added the specific plugins to `options.json`. How can I fix this?

bettiolo commented:
I have the following failing test and I am not sure how to resolve:

```
1 failing

  1) babel-preset-es2017/es2017 exec:
     ReferenceError: /Users/mbettiolo/Code/github/babel/packages/babel-preset-es2017/test/fixtures/es2017/exec.js: require is not defined   1 | 'use strict';
   2 |
   3 | var _getOwnPropertyDescriptors = require('babel-runtime/core-js/object/get-own-property-descriptors');
```

danez commented:
Imho the presets don’t necessarily need tests, because the plugins already have tests. Same for the polyfills, as core-js has already tests.
The only exception should be preset-options now (if used), to see if they enable/disable/change the correct plugins.

hzoo commented:
I think we would like to go with something like https://github.com/babel/babel/pull/3624 + update babel-polyfill so you can import es2015, 2016, etc separately for more granularity.

I know the current preset on npm has the polyfills built-in but I'm not sure we all agree on doing it (can always add it in at a later point if do). What I would suggest is making another preset like `babel-preset-es2017-with-polyfill`. Either way we'll have to update the current version.

Users currently don't expect the polyfills to be there, or some of the other transforms currently in the preset so I think it would be good to publish a release with only the transforms in es2017 as soon as we can because it currently does something different than `babel-preset-es2015` and es2016 (also that it seems like an "official" preset but it isn't right now).

hzoo commented:
@bettiolo pinging again - I think we want to move forward with something like https://github.com/babel/babel/pull/3624 and then to modify babel-polyfill to support polyfilling each year like in babel-polyfill/es2015, etc.

Would it be too much effort to publish your module/preset with another name (babel-preset-es2017-with-polyfill, etc), and giving babel the `es2017` name? Like I tried to explain I think some users might be expecting something different from what `babel-preset-es2017` currently offers given it's a similar name to `babel-preset-es2015`, 2016, etc.

We can also revisit adding the polyfill under the preset since atm I don't think we all agree on adding it right now.

hzoo commented:
@bettiolo sorry for the spam, I don't know if you're busy but I would like to hear your thoughts on ^. Again I don't think this PR is what we want so we might just go with https://github.com/babel/babel/pull/3624 and have to publish under a different name.

bettiolo commented:
Made the ownership change, please bump major.
